ui/src/redux/thunks/user.js

```diff
@@ -27,3 +34,23 @@ export const getUserPendingMembers = () => async (dispatch, getState) => {
         dispatch(loadUserPendingMembers(userPendingMembersList, expiry));
     }
 };
+
+export const getUserResourceAccessList = () => async (dispatch, getState) => {
+    if (getState().user.userResourceAccessList) {
```

Comment:
I am not sure if userResourceAccessList can be updated but if it is it should come after the check of expiry part, because if you put it here it will never be expired and refreshed

Reply:
good point, updated logic to first check for expiry then fetch data if needed.